### ISSUE TITLE
Support for oAuth 2.0 Introspection Plugin

### ISSUE THREAD

**Issue by rajarameshvarma** [pantsel/konga#231]
Is there any plan to implement support for "oAuth 2.0 Introspection" plugin. Also enhanced features for the Kong Enterprise only plugin's like Rate Limiting etc.

I know Kong Enterprise comes with its own Admin ports, but unfortunately that lacks in many areas and you are definitely miles a head. Thank you for that.

**Comment by pantsel** [comment]
@rajarameshvarma , thaks for your kind words.

It would be nice to implement premium Kong plugins in Konga but unfortunatelly I don't know how they work nor their schema.

I haven't got the chance to test things out with a Kong EE instance.

Have you tried connecting Konga to a Kong EE instance?

Don't the premium plugins and the enchanced features show up on Konga's lists? (Even us custom plugins)

If they don't appear automatically, it means that they are probably implemented in a different manner.

**Comment by rajarameshvarma** [comment]
I haven't tried connecting Konga to a Kong EE instance. I will update the thread once I get a chance to check. Thank you for your quick response.

**Comment by mynameisgv** [comment]
I tried connecting konga to a enterprise version 0.33. It looks very different.
it shows the available plugins. But does not show routes.

![image](https://user-images.githubusercontent.com/6636219/43625235-cf41a2b2-96a0-11e8-8e67-b952dbd4c190.png)

**Comment by pantsel** [comment]
This version of konga is old

**Comment by mynameisgv** [comment]
@pantsel this is version KONGA 0.12.0. latest in dockerhub.
The problem is not with konga. when i connect to free version routes do show up.
0.33 enterprise version do not have the services and routes feature yet.
https://docs.konghq.com/enterprise/0.33-x/getting-started/adding-your-api/

**Comment by pantsel** [comment]
@mynameisgv , indeed. Didn't knew that. Never stumbled across a EE of Kong.
Apart from that, the versioning is totally different so features that depend on CE versioning, will not show or work as expected.

**Comment by mynameisgv** [comment]
@pantsel  i tested konga with enterprise version. 
most of the things work.
routes do not show up because of  the way free version and enterprise version are tagged. 
0.13.1
0.33-enterprise-edition

 var fver = version.split('.').slice(0, -1).join('');
and 01_layout-controllers.js for example blocks routes in the menu based on this version.
return AuthService.hasPermission('services', 'read') && $rootScope.isGatewayVersionEqOrGreater('0.13.0')

version.split('.').slice(0, -1).join(''); this returns 0 for enterprise edition.
But if i juut use the url http://localhost/#!routes things work fine.

Is this check necessary? 

I am still testing other functionality.

**Comment by pantsel** [comment]
Yes it is. Konga is designed to work with tge CE edition and the version checks are mandatory

**Comment by spipriya** [comment]
Couple of observations if it helps anyone using Kong Enterprise. 
1. Kong EE Plugins (openID connect etc) show up as Custom plugins in Konga. 
2. Because of version tag check, routes & services are disabled on UI. However it can still be accessed via #!/routes & /#!/services and seems to work fine.
